fuzz/src/main/java/edu/berkeley/cs/jqf/fuzz/junit/quickcheck/FuzzStatement.java

```diff
@@ -170,6 +180,26 @@ public void evaluate() throws Throwable {
                         failures.add(e);
                     }
                 }
+                long endTrialTime = System.currentTimeMillis();
+                if (System.getProperty("jqfObservability") != null) {
```

Comment:
Is it possible to add a new function in `Guidance` called `observeGuidance` or something. And each Guidance (`ZestGuidance`, `NoGuidance`, etc) implements its own `observeGuidance` method. 

This allows the compiler to enforce the implementation of `observeGuidance` when new guidance is implemented. It also avoids type checks here.

Reply:
I was a bit unsure about this since optional logging doesn't really warrant a change to the `Guidance` interface, but I added a `default` method